src/features/todos.ts

```diff
@@ -1,8 +1,33 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 import { Todo } from '../types/Todo';
 
-export const todosSlice = createSlice({
+export interface TodosState {
+  data: Todo[];
+  loading: boolean;
+  error: string | null;
+}
+
+const initialState: TodosState = {
+  data: [],
+  loading: false,
+  error: null,
+};
+
+const todosSlice = createSlice({
   name: 'todos',
-  initialState: [] as Todo[],
-  reducers: {},
+  initialState,
+  reducers: {
+    initTodos: (state, action: PayloadAction<Todo[]>) => {
+      return Object.assign({}, state, { data: action.payload });
```

Comment:
```suggestion
      return {...state, data: action.payload};
```
we can simplify that